FabricObserver/Observers/AppObserver.cs

```diff
@@ -2057,13 +2189,65 @@ private Task<ParallelLoopResult> MonitorDeployedAppsAsync(CancellationToken toke
+                    
+                    if (procId == parentPid)
+                    {
+                        AllAppPrivateBytesDataMb[id].AddData(memPb);
+
+                        // RG
+                        if (rgMemoryPercentThreshold > 0 && repOrInst.RGMemoryLimitMb > 0)
+                        {
+                            double pct = ((double)memPb / repOrInst.RGMemoryLimitMb) * 100;
+                            AllAppRGMemoryUsagePercent[id].AddData(pct);
+                        }
+                    }
+                    else
+                    {
+                        _ = AllAppPrivateBytesDataMb.TryAdd(
+                               $"{id}:{procName}{procId}",
+                               new FabricResourceUsageData<float>(
+                                       ErrorWarningProperty.PrivateBytesMb,
+                                       $"{id}:{procName}{procId}",
+                                       capacity,
+                                       UseCircularBuffer,
+                                       EnableConcurrentMonitoring));
+                        
+                        AllAppPrivateBytesDataMb[$"{id}:{procName}{procId}"].AddData(memPb);
+                        
+                        // RG
+                        if (rgMemoryPercentThreshold > 0 && repOrInst.RGMemoryLimitMb > 0)
+                        {
+                            _ = AllAppRGMemoryUsagePercent.TryAdd(
+                                $"{id}:{procName}{procId}",
+                                new FabricResourceUsageData<double>(
+                                        ErrorWarningProperty.RGMemoryUsagePercent,
+                                        $"{id}:{procName}{procId}",
+                                        capacity,
+                                        UseCircularBuffer,
+                                        EnableConcurrentMonitoring));
+                            
+                            double pct = ((double)memPb / repOrInst.RGMemoryLimitMb) * 100;
+                            AllAppRGMemoryUsagePercent[$"{id}:{procName}{procId}"].AddData(pct);
+                        }
+                    }
+                }
+
+                // Working Set.
                 if (checkMemMb)
                 {
                     if (procId == parentPid)
                     {
-                        _ = ProcessInfoProvider.Instance.GetProcessWorkingSetMb(procId, procName, Token, checkPrivateWS);
-                        Thread.Sleep(100);
-                        AllAppMemDataMb[id].AddData(ProcessInfoProvider.Instance.GetProcessWorkingSetMb(procId, procName, Token, checkPrivateWS));
+                        if (usePerfCounter)
+                        {
+                            _ = ProcessInfoProvider.Instance.GetProcessWorkingSetMb(procId, procName, Token, usePerfCounter);
+                            Thread.Sleep(100);
```

Comment:
do we need to use sleep while checking private bytes above as well?

Reply:
Since this is a one time check and no perf counter is being warmed up, I don't think so. It doesn't add any value.